tee-worker/ts-tests/integration-tests/assertion_contracts.test.ts

```diff
@@ -0,0 +1,132 @@
+import { randomBytes, KeyObject } from 'crypto';
+import { step } from 'mocha-steps';
+import { initIntegrationTestContext } from './common/utils';
+import { assertIsInSidechainBlock, assertVc } from './common/utils/assertion';
+import {
+    getSidechainNonce,
+    getTeeShieldingKey,
+    sendRequestFromTrustedCall,
+    createSignedTrustedCallRequestVc,
+} from './common/di-utils'; // @fixme move to a better place
+import type { IntegrationTestContext } from './common/common-types';
+import { aesKey } from './common/call';
+import type { CorePrimitivesIdentity } from 'parachain-api';
+import type { LitentryValidationData, Web3Network } from 'parachain-api';
+import type { Vec, Bytes } from '@polkadot/types';
+import fs from 'fs';
+import path from 'path';
+import { assert } from 'chai';
+import { genesisSubstrateWallet } from './common/helpers';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { subscribeToEvents, subscribeToEventsWithExtHash } from './common/transactions';
+import { encryptWithTeeShieldingKey } from './common/utils/crypto';
+import { ethers } from 'ethers';
+import { sleep } from './common/utils';
+import { $ as zx } from 'zx';
+
+describe('Test Vc (direct request)', function () {
+    let context: IntegrationTestContext = undefined as any;
+    let teeShieldingKey: KeyObject = undefined as any;
+    let aliceSubstrateIdentity: CorePrimitivesIdentity = undefined as any;
+
+    let alice: KeyringPair = undefined as any;
+    let contractBytecode = undefined as any;
+    const clientDir = process.env.LITENTRY_CLI_DIR;
+
+    this.timeout(6000000);
+
+    before(async () => {
+        context = await initIntegrationTestContext(
+            process.env.WORKER_ENDPOINT!, // @fixme evil assertion; centralize env access
+            process.env.NODE_ENDPOINT! // @fixme evil assertion; centralize env access
+        );
+        teeShieldingKey = await getTeeShieldingKey(context);
+        aliceSubstrateIdentity = await context.web3Wallets.substrate.Alice.getIdentity(context);
+        alice = genesisSubstrateWallet('Alice');
+    });
+
+    step('loading tokenmapping contract bytecode', async function () {
+        const file = path.resolve('./', './contracts-build-info/TokenMapping.sol/TokenMapping.json');
```

Comment:
I know `TokenMapping.sol` is generated out by compiler. But originally how does it come out? derived from which sol file? or ?

Next question would be: how easy/difficult to extend this test structure to cover any other solidity files?

Reply:
>I know TokenMapping.sol is generated out by compiler. But originally how does it come out? derived from which sol file? or ?

TokenMapping.sol comes from [TokenMapping.sol](https://github.com/litentry/litentry-parachain/blob/dev/tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/TokenMapping.sol), the foundry compilation will generate according to the contract name.

>Next question would be: how easy/difficult to extend this test structure to cover any other solidity files?
- We could easily add tests for other token contracts, such as directly adding `['bnb', 'ada'...] `to test.
- Regarding internal testing of the contract , we could write based on Foundry's tests, which is actually not difficult. However, this PR tends to focus more on deploying contract tests.